client/rpc/src/eth/filter.rs

```diff
@@ -118,19 +119,19 @@ where
 	BE: Backend<B> + 'static,
 	BE::State: StateBackend<BlakeTwo256>,
 {
-	fn new_filter(&self, filter: Filter) -> Result<U256> {
+	fn new_filter(&self, filter: Filter) -> RpcResult<U256> {
 		self.create_filter(FilterType::Log(filter))
 	}
 
-	fn new_block_filter(&self) -> Result<U256> {
+	fn new_block_filter(&self) -> RpcResult<U256> {
 		self.create_filter(FilterType::Block)
 	}
 
-	fn new_pending_transaction_filter(&self) -> Result<U256> {
+	fn new_pending_transaction_filter(&self) -> RpcResult<U256> {
 		Err(internal_err("Method not available."))
 	}
 
-	fn filter_changes(&self, index: Index) -> BoxFuture<Result<FilterChanges>> {
+	async fn filter_changes(&self, index: Index) -> RpcResult<FilterChanges> {
```

Comment:
We should be able to simplify this fn

Reply:
The same, don't think that we should to do it in this PR.

It's better to have a good review later and decide about possible improvements.